core/src/main/java/com/expedia/www/haystack/client/Tracer.java

```diff
@@ -0,0 +1,216 @@
+            this.references = new ArrayList<>();
+            this.tags = new HashMap<>();
+        }
+
+		@Override
+		public SpanBuilder asChildOf(io.opentracing.SpanContext parent) {
+        return this.addReference(References.CHILD_OF, parent);
+		}
+
+		@Override
+		public SpanBuilder asChildOf(BaseSpan<?> parent) {
+        if (parent == null) {
+            return this;
+        }
+        return this.addReference(References.CHILD_OF, parent.context());
+		}
+
+		@Override
+		public SpanBuilder addReference(String referenceType, io.opentracing.SpanContext referencedContext) {
+        if (referencedContext == null) {
+            return this;
+        }
+
+        if (!(referencedContext instanceof SpanContext)) {
+            // can't do much here, so ignore it
+            return this;
+        }
+
+        references.add(new Reference(referenceType, (SpanContext) referencedContext));
+        return this;
+		}
+
+		@Override
+		public SpanBuilder ignoreActiveSpan() {
+        this.ignoreActive = true;
+        return this;
+		}
+
+      @Override
+      public SpanBuilder withTag(String key, String value) {
+          this.tags.put(key, value);
+          return this;
+      }
+
+      @Override
+      public SpanBuilder withTag(String key, boolean value) {
+          this.tags.put(key, value);
+          return this;
+      }
+
+      @Override
+      public SpanBuilder withTag(String key, Number value) {
+          this.tags.put(key, value);
+          return this;
+      }
+
+		@Override
+		public SpanBuilder withStartTimestamp(long microseconds) {
+        this.startTime = microseconds;
+        return this;
+		}
+
+		@Override
+		public ActiveSpan startActive() {
+        return tracer.makeActive(startManual());
+		}
+
+      private SpanContext createNewContext() {
+          UUID randomId = UUID.randomUUID();
+          UUID zero = new UUID(0l, 0l);
+          return new SpanContext(randomId, randomId, zero);
+      }
+
+      private SpanContext createDependantContext() {
+          Reference parent = references.get(0);
+          for (Reference reference : references) {
+              if (References.CHILD_OF.equals(reference.getReferenceType())) {
+                  // first parent wins
+                  parent = reference;
+                  break;
+              }
+          }
+
+          Map<String, String> baggage = new HashMap<>();
+          for (Reference reference : references) {
+              baggage.putAll(reference.getContext().getBaggage());
+          }
+
+          return new SpanContext(parent.getContext().getTraceId(),
+                                 UUID.randomUUID(),
+                                 parent.getContext().getSpanId(),
+                                 baggage);
+      }
+
+      private SpanContext createContext() {
+          if (references.isEmpty()) {
+              return createNewContext();
+          }
+          return createDependantContext();
```

Comment:
dependent (emphasis on the third 'e'). Yeah, I won the junior high spelling bee.